Src/Plugins/Security/PFXCryptographicKeyFileValidator.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    internal static class PFXCryptographicKeyFileValidator
+    {
+#pragma warning disable IDE0060 // Remove unused parameter
+        public static string IsValid(
+            ref string matchedPattern,
+            ref Dictionary<string, string> groups,
+            ref bool performDynamicValidation,
+            ref string failureLevel,
+            ref string fingerprint)
+        {
+#pragma warning restore IDE0060
+            fingerprint = $"[cert={matchedPattern}]";
+
+            // This plugin does not perform any dynamic validation.
+            // We therefore set this setting to false. This is a
+            // clue to the caller not to warn the user that, e.g.,
+            // dynamic analysis was available but not exercised.
+            performDynamicValidation = false;
+
+            return TryLoadCertificate(matchedPattern).ToString();
+        }
+
+        private static ValidationState TryLoadCertificate(string certificatePath)
+        {
+            X509Certificate2 certificate = null;
+            try
+            {
+                // If this certificate needs a password or it is a bundle, it will throw an exception.
+                certificate = new X509Certificate2(certificatePath);
+
+                return certificate.PrivateKey != null ? ValidationState.Authorized : ValidationState.NoMatch;
```

Comment:
Sweet! the only problem here is that you're not describing the asset (in this case the public key token of the certificate). The NoMatch return is interesting here! Generally, we drop NoMatch items entirely on the floor. This may confuse people for this case. We might want to consider a new validation state for killed or incomplete or public only credentials. another option would be to raise a NotApplicable message, this check isn't applicable in this case because we're got a public key only result. Stick with NoMatch right now, but let's file a tracking issue.